### PR TITLE
Add local mode for v1 primitives

### PR DESCRIPTION
**Pull request by jyu00**
### Summary

Support a "local testing mode" by allowing fake and Aer backends passed to primitives. This currently only works with V1 primitive because there is no BackendPrimitive for V2 yet. 

### Details and comments
Fixes #

### CODE REVIEW

File: releasenotes/notes/local-test-v1-ad62507880934b29.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+features:
+  - |
+    A new local testing mode is added. It allows you to
+    validate your quantum prorams before sending them to a physical system.
```
Comment (kt474) [inline releasenotes/notes/local-test-v1-ad62507880934b29.yaml:5]:
```suggestion
    validate your quantum programs before sending them to a physical system.
```

File: test/unit/test_ibm_primitives_v2.py [hunk 2]
```diff
@@ -218,13 +218,15 @@ def test_init_with_no_backend_session_quantum(self, primitive):
     @data(EstimatorV2, SamplerV2)
     def test_default_session_context_manager(self, primitive):
         """Test getting default session within context manager."""
-        service = MagicMock()
-        backend = "ibm_gotham"
+        backend_name = "ibm_gotham"
+        backend = get_mocked_backend(name=backend_name)
+        # service = MagicMock()
+        # backend = "ibm_gotham"
```
Comment (kt474) [inline test/unit/test_ibm_primitives_v2.py:224]:
```suggestion
```

**Review by kt474 — APPROVED** [review]
LGTM, minor typos

**Comment by maniraman-periyasamy** [comment]
Does the local mode supports error-mitigation such as ZNE (resilience_level=2). I could not find this information in the documentation.

**Pull request merged**